client/packages/common/src/hooks/useElectronClient/useElectronClient.ts

```diff
@@ -88,7 +88,7 @@ export const useElectronClient = (discover = false) => {
   }, []);
 
   useEffect(() => {
-    if (timedOut && state.servers.length == 0) {
+    if (timedOut && !state.discoveryTimedOut && state.servers.length == 0) {
```

Comment:
setting the state to be a new object triggers the effect, which sets the state to a new object..

Reply:
ohh, forgot you shouldn't really mutate the thing that's in the trigger. I wonder if you we need to listen to `state` in this useEffect at all ?

Reply:
I expected that react would be able to cope, a shallow match should be able to determine similarity and not re-render. but apparently not 🤷 

Still, I like your suggestion of removing `state` better, so have popped that in. I've simplified slightly which results in two extra renders, but the code is nicer 😉